### PR TITLE
Check to make sure it is a Pod object

### PR DESCRIPTION
**Pull request by liwenwu-amazon**
*Issue #168, :*

*Description of changes:*
Check to make sure the object is a Pod type

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/k8sapi/discovery.go [hunk 1]
```diff
@@ -208,7 +208,7 @@ func (d *Controller) handlePodUpdate(key string) error {
 	}
 
 	pod, ok := obj.(*v1.Pod)
-	if !ok {
+	if !ok || pod == nil {
```
Comment (nckturner) [inline pkg/k8sapi/discovery.go]: What circumstances cause pod to be nil?

Comment (liwenwu-amazon) [reply pkg/k8sapi/discovery.go]: @nckturner it happens once in @rtripat setup once.  I am not able to reproduce it.

Comment (nckturner) [reply pkg/k8sapi/discovery.go]: Ok well it looks to me like it might help prevent #168 but I'm wondering why its receiving `nil` in the first place.

Comment (mattlandis) [reply pkg/k8sapi/discovery.go]: @ofiliz 

I think this is OK.
A nil interface is != nil but a nil interface converted to  struct is nil and pod is a struct.

https://play.golang.org/p/uAZJG2_mUuL

Unless I am missing something.

Comment (mattlandis) [reply pkg/k8sapi/discovery.go]: lgtm

I agree with nic that we should investigate further why we can get a nil pod here if exists is true.  Somewhere we are storing nil in the datastore.

**Pull request merged**